tensornetwork/backends/jax/jax_backend.py

```diff
@@ -18,20 +18,23 @@
 from typing import Any, Optional
 from tensornetwork.backends.numpy import numpy_backend
 import numpy
+
+
 Tensor = Any
 
 
 class JaxBackend(numpy_backend.NumPyBackend):
   """See base_backend.BaseBackend for documentation."""
 
+  try:
```

Comment:
When would this code run? When the file is imported or when the class is used?

Reply:
Hmmm, I'm confused. From the top of my mind, I'd say when the file is imported, right? Unless there's a diff python2-3 I'm not aware of. But his should cause tests too fail, are we still missing a test, or do the tests not work?

Reply:
Ok nevermind, tests work but we should just add more tests of the ones you advised. Not just test import tensornetwork but all relevant submodules. I'll add those.

However, I have a small issue now. The reason I (mistakenly) moved them out of the init is that for pytorch, the typing in init: `def __init__(self, dtype: Optional[torch.dtype] = None):` forces me to already know torch...

Reply:
Just make them `Any`s.

Reply:
Just importing `tensornetwork` should be fine, don't worry about full module converage since all of the things the user is expected to have direct access to is already imported in `__init__`

Reply:
Except for the backends! And that would have caught the bug you noticed. I also added most important submodules, doesn't hurt, right?

Reply:
Might be annoying if we do a refactor, but refactors are annoying anyway so it's fine.

Reply:
Actually, let me check what is already covered by the init of tensornetwork and the other tests and put it to a minimum, no need introducing extra annoyances